### ISSUE TITLE
Version 11+ makes Windows 10 unusable / memory leak 

### ISSUE THREAD

**Issue by pocesar** [nanocurrency/raiblocks#748]
**Description of bug:**

Wallets starts briefly, then freezes, database size stays the same

![image](https://user-images.githubusercontent.com/461084/37852812-c89abef8-2ec2-11e8-936d-3a48d4059fe7.png)

**Steps to reproduce the issue:**
1. Nano Milestone 11 on Windows on a fresh start (deleted database files)
2. Main window hangs and werfault.exe process starts using 100% of CPU (on one core)

**Describe the results you received:**

Windows Error Reporting starts using 100% (on one core), and is unkillable.
![image](https://user-images.githubusercontent.com/461084/37852573-e58120b2-2ec1-11e8-8d05-97bf8cfbcf79.png)
![image](https://user-images.githubusercontent.com/461084/37852645-2560c188-2ec2-11e8-82fa-483b5a076cfc.png)

the only way to stop this is forcefully restarting the computer (taskkill can't kill it)

![image](https://user-images.githubusercontent.com/461084/37852719-68301e64-2ec2-11e8-98dd-17a48a63868f.png)

![image](https://user-images.githubusercontent.com/461084/37852741-8918b2bc-2ec2-11e8-9df8-518319629055.png)

**Describe the results you expected:**

It works like previous versions

**Additional information you deem important (e.g. issue happens only occasionally):**

Happened every time, from the 3 times I tried

**Environment**:

- OS information: Windows 10 x64
- Node version: Milestone 11

**logs**

```
[2018-03-23 17:37:44.065215]: Bootstrap stopped because there are no peers
[2018-03-23 17:37:44.065215]: Bootstrap stopped because there are no peers
[2018-03-23 17:37:44.065215]: Bootstrap stopped because there are no peers
[2018-03-23 17:37:44.065215]: Bootstrap stopped because there are no peers
[2018-03-23 17:37:44.065215]: Exiting bootstrap attempt
[2018-03-23 17:37:44.510075]: Beginning pending block search
[2018-03-23 17:37:44.510075]: Pending block search phase complete
[2018-03-23 17:37:45.515069]: UDP Receive error: No connection could be made because the target machine actively refused it
[2018-03-23 17:37:47.737077]: Beginning pending block search
[2018-03-23 17:37:47.737077]: Pending block search phase complete
[2018-03-23 17:37:49.068256]: Starting bootstrap attempt
[2018-03-23 17:37:49.238260]: Connection established to [::ffff:192.99.176.122]:7075
[2018-03-23 17:37:49.246260]: Connection established to [::ffff:192.99.176.121]:7075
[2018-03-23 17:37:49.246260]: Connection established to [::ffff:165.227.201.217]:7075
[2018-03-23 17:37:49.248260]: Connection established to [::ffff:144.217.167.119]:7075
[2018-03-23 17:37:49.274260]: Connection established to [::ffff:159.89.143.80]:7075
[2018-03-23 17:37:49.276261]: Connection established to [::ffff:138.68.2.234]:7075
[2018-03-23 17:37:49.304261]: Connection established to [::ffff:139.162.199.142]:7075
[2018-03-23 17:37:49.322262]: Connection established to [::ffff:138.201.94.249]:7075
[2018-03-23 17:37:49.402263]: Invalid size: expected 64, got 0
[2018-03-23 17:37:49.404263]: frontier_req failed, reattempting
[2018-03-23 17:37:59.094699]: Error initiating bootstrap connection to [::ffff:89.64.59.63]:10025: The I/O operation has been aborted because of either a thread exit or an application request
[2018-03-23 17:37:59.094699]: Error initiating bootstrap connection to [::ffff:192.81.216.141]:7075: The I/O operation has been aborted because of either a thread exit or an application request
[2018-03-23 17:37:59.094699]: Error initiating bootstrap connection to [::ffff:54.246.128.136]:7075: The I/O operation has been aborted because of either a thread exit or an application request
[2018-03-23 17:37:59.350706]: Connection established to [::ffff:45.76.92.115]:7075
[2018-03-23 17:37:59.352705]: Connection established to [::ffff:212.47.237.7]:7075
[2018-03-23 17:37:59.356701]: Connection established to [::ffff:188.226.155.250]:7075
[2018-03-23 17:37:59.484704]: Connection established to [::ffff:172.104.32.150]:7075
[2018-03-23 17:38:01.985391]: Found a representative at [2001:41d0:8:d85f::1]:7075
[2018-03-23 17:38:02.857584]: Found a representative at [2600:3c03::f03c:91ff:fee5:29e]:7075
[2018-03-23 17:38:04.403929]: Received 303959 frontiers from [::ffff:192.99.176.121]:7075
[2018-03-23 17:38:09.642300]: Found a representative at [::ffff:139.59.31.249]:7075
[2018-03-23 17:38:10.134317]: Found a representative at [::ffff:35.200.122.8]:7075
[2018-03-23 17:38:10.548320]: Completed frontier request, 468710 out of sync accounts according to [::ffff:192.99.176.121]:7075
[2018-03-23 17:38:10.931790]: Found a representative at [::ffff:178.22.66.84]:7075
[2018-03-23 17:38:10.932791]: Found a representative at [::ffff:178.22.66.84]:7075
[2018-03-23 17:38:11.020796]: Requesting account xrb_33de47sgua1pup78kw4ye8nz7ujdz47oucfgco1zb95n8n7hkbq1x81z3md6 from [::ffff:165.227.201.217]:7075. 468709 accounts in queue
[2018-03-23 17:38:11.020796]: Error receiving block type End of file
[2018-03-23 17:38:11.021797]: Error receiving block type End of file
```

**Comment by argakiig** [comment]
Had you run a previous build? Did you remove everything from the data dir or just the data.ldb and could you restart the computer, clear out the data dir and try again if you haven't already

**Comment by pocesar** [comment]
yes, since version 8, but since starting milestone 11 crashed, I decided to wipe the databases and wallet files

**Comment by argakiig** [comment]
Can you try this version instead and let me know what the results, You should uninstall the previous and reinstall with this one.
https://ci.appveyor.com/api/buildjobs/lvlgfxfjamx5cdlc/artifacts/Nano_Installer-11.0.GIT-a22ceb5-win64.exe

**Comment by pocesar** [comment]
sorry for the delay on the response. I'm trying 11.1 right now and will report back. so far, it says it's synchronizing without the lockup, and using my previous seed without any issue

![image](https://user-images.githubusercontent.com/461084/38183651-80374750-3618-11e8-92d2-c23da6e18f43.png)

**Comment by pocesar** [comment]
ok, it seems that it's stuck forever on Block 1, even from a fresh start. the block count keep growing, the database file is 1,7GB, but it's just "frozen" in time, although the GUI is responsive.

issue #635 still happens (closed it, and the process is in the background doing nothing, but using more and more memory each time it passes)

![image](https://user-images.githubusercontent.com/461084/38200808-25b68e68-366c-11e8-9b1c-0ea600a6e867.png)

**Comment by argakiig** [comment]
what did the unchecked count get to? chances are the blocks needing to be processed to continue with the bootstrapping had not been received yet and as a result would act as a blocker to checking in further blocks. The process still running after exit is still being investigated

**Comment by pocesar** [comment]
it was around 3,8kk and the process is still stuck , and eventually will memory leak until my system crashes

![image](https://user-images.githubusercontent.com/461084/38201927-61535e0c-3670-11e8-9c6d-b022b2cae08f.png)

**Comment by argakiig** [comment]
Does the memory leak only start when you exit?

**Comment by pocesar** [comment]
yes, while it was opened, the memory was stable, but the UI wasn't doing anything

**Comment by argakiig** [comment]
also 3.8kk is only about half downloaded, that would explain why you werent getting past 1 if it hadnt downloaded the block yet. You may want to wait for all almost 7.3m blocks to download as it could be the very last block you download that is needed to continue checking in blocks

**Comment by pocesar** [comment]
well, that's really counter productive. anyway, the process is memory leaking and nearing 2GB, I'll have to kill it

![image](https://user-images.githubusercontent.com/461084/38202528-69306064-3672-11e8-9780-8b884d7e0387.png)

**Comment by clemahieu** [comment]
It sounds like you’re running it on a machine with slow disk IO. Ignore it can't write to disk fast enough it keeps it in memory.

Improving disk IO is slowly coming to the top of the list of things to work on, there have been other higher priority issues.

**Comment by pocesar** [comment]
I'm on a Samsung EVO 850 👍

**Comment by clemahieu** [comment]
Odd, then I haven’t seen a memory spike like that, we’ll have to see what it takes to reproduce.

**Comment by argakiig** [comment]
how much memory does it show as using while its running and stable?

**Comment by pocesar** [comment]
@argakiig this is the usual memory usage

![image](https://user-images.githubusercontent.com/461084/38227080-d8b731f0-36d2-11e8-9311-acf451143949.png)

that's why it's a clear indication of a serious memory leak

**Comment by clemahieu** [comment]
It’s be interesting to know what it stabilizes to. When I run it on OSX it’s pretty stable around 70-120. I’ve seen it spike on systems that have bad disk IO as it puts blocks in memory until written to disk but with that SSD that’s clearly not the case.

**Comment by pocesar** [comment]
yeah, the I/O isn't high, that's why it's strange, maybe it's an Windows issue, I decided to monitor the disk activity and it seems fine, no crazy reads or writes, it tops around 10MB/s, averaging 2MB/s, so it's clearly not an I/O issue. average response time is 3ms. it's fairly constant as expected, the only issue is when you close the wallet

![image](https://user-images.githubusercontent.com/461084/38239300-b407031e-3702-11e8-8b4f-c6737bc7a035.png)

![image](https://user-images.githubusercontent.com/461084/38239327-c48afa74-3702-11e8-99e2-acbd5a19a25a.png)

![image](https://user-images.githubusercontent.com/461084/38239346-cff24ef8-3702-11e8-980b-1557cfa29ab9.png)

don't know what else to look for. I'm in a older system besides the SSD though (Core 2 Quad Q9550 and 8 GB of DDR2 800) but have no issues with, for example, Bitcoin Core or other more intensive / aggressive / non-optimized coins, like Ethereum

**Comment by rkeene** [comment]
This should be resolved by the memory leak fixes supplied in V17.0 !  See #1338 for additional details.
